### DISCUSSION TITLE
Why do we need DBCP for Redis Storage + Lettuce?

### DISCUSSION THREAD

**Discussion by jeevangali-agi** [jobrunr/jobrunr#201] [Q&A] [answered]
Can you please explain the below the content from Documentation. why do we need DBCP?
Lettuce says, it is thread safe.

https://mashhurs.wordpress.com/2020/03/26/jedis-vs-lettuce-java-redis-clients/#:~:text=TL%3BDR%3A%20The%20Lettuce%20provides%20multiple%20threads%2C%20and%20required,are%20able%20to%20manage%20process%20%28get%20or%20destroy%29.

![image](https://user-images.githubusercontent.com/73397163/130358522-92c77c09-890b-4aad-8547-565476cc7384.png)

**Comment by rdehuyss** [comment, marked as answer]
See https://lettuce.io/core/release/reference/#connection-pooling.is-connection-pooling-necessary

7.10.1. Is connection pooling necessary?
Lettuce is thread-safe by design which is sufficient for most cases. All Redis user operations are executed single-threaded. Using multiple connections does not impact the performance of an application in a positive way. The use of blocking operations usually goes hand in hand with worker threads that get their dedicated connection. **The use of Redis Transactions is the typical use case for dynamic connection pooling as the number of threads requiring a dedicated connection tends to be dynamic.** That said, the requirement for dynamic connection pooling is limited. Connection pooling always comes with a cost of complexity and maintenance.

> JobRunr uses Redis transactions, that's why it is needed.

See also https://github.com/lettuce-io/lettuce-core/issues/1832
